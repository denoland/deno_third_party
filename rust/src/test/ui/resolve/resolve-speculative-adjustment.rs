// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
// file at the top-level directory of this distribution and at
// http://rust-lang.org/COPYRIGHT.
//
// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
// option. This file may not be copied, modified, or distributed
// except according to those terms.

// Make sure speculative path resolution works properly when resolution
// adjustment happens and no extra errors is reported.

struct S {
    field: u8,
}

trait Tr {
    fn method(&self);
}

impl Tr for S {
    fn method(&self) {
        fn g() {
            // Speculative resolution of `Self` and `self` silently fails,
            // "did you mean" messages are not printed.
            field;
            //~^ ERROR cannot find value `field`
            method();
            //~^ ERROR cannot find function `method`
        }

        field;
        //~^ ERROR cannot find value `field`
        method();
        //~^ ERROR cannot find function `method`
    }
}

fn main() {}
