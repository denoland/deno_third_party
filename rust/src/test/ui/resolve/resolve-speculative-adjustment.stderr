error[E0425]: cannot find value `field` in this scope
  --> $DIR/resolve-speculative-adjustment.rs:27:13
   |
LL |             field;
   |             ^^^^^ not found in this scope

error[E0425]: cannot find function `method` in this scope
  --> $DIR/resolve-speculative-adjustment.rs:29:13
   |
LL |             method();
   |             ^^^^^^ not found in this scope

error[E0425]: cannot find value `field` in this scope
  --> $DIR/resolve-speculative-adjustment.rs:33:9
   |
LL |         field;
   |         ^^^^^ help: try: `self.field`

error[E0425]: cannot find function `method` in this scope
  --> $DIR/resolve-speculative-adjustment.rs:35:9
   |
LL |         method();
   |         ^^^^^^ help: try: `self.method`

error: aborting due to 4 previous errors

For more information about this error, try `rustc --explain E0425`.
